modules/nf-core/diamond/makedb/main.nf

```diff
@@ -1,31 +1,52 @@
 process DIAMOND_MAKEDB {
-    tag "$fasta"
+    tag "$meta.id"
     label 'process_medium'
 
     conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/diamond:2.0.15--hb97b32f_0' :
-        'biocontainers/diamond:2.0.15--hb97b32f_0' }"
+        'https://depot.galaxyproject.org/singularity/diamond:2.1.8--h43eeafb_0' :
+        'biocontainers/diamond:2.1.8--h43eeafb_0' }"
 
     input:
-    path fasta
+    tuple val(meta), path(fasta)
 
     output:
-    path "${fasta}.dmnd", emit: db
-    path "versions.yml" , emit: versions
+    tuple val(meta), path("*.dmnd"), emit: db
+    path "versions.yml"            , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def is_compressed = fasta.name.endsWith(".gz")
+    def fasta_name = fasta.name.replace(".gz", "")
+
     """
+    if [ "${is_compressed}" == "true" ]; then
+        gzip -c -d ${fasta} > ${fasta_name}
```

Comment:
I can't remember exactly... Can makedb only accept a single fasta file or Multiple? Would this work in the latter case?

Reply:
It only accepts one fasta file, so we are ok with this